registration-processor/registration-processor-core/src/test/java/io/mosip/registration/processor/core/eventbus/KafkaMosipEventBusTest.java

```diff
@@ -1,627 +1,668 @@
+			Mockito.when(asyncResultForMessageDTO.succeeded()).thenReturn(false);
+			Mockito.when(asyncResultForMessageDTO.cause()).thenReturn(new MessageExpiredException());
+			((Handler<AsyncResult<MessageDTO>>) arguments.getArgument(1))
+				.handle(asyncResultForMessageDTO);
+			if (!async.isCompleted())
+				async.complete();
+            
+            return null;
+		}).when(eventHandler).handle(any(), any());
+
+		kafkaMosipEventBus.consume(MessageBusAddress.PACKET_VALIDATOR_BUS_IN, eventHandler);
+		async.await();
+
+		verify(eventHandler, times(testDataCount)).handle(any(), any());
+		verify(kafkaConsumer, times(1)).pause(
+			any(io.vertx.kafka.client.common.TopicPartition.class), any());
+		verify(kafkaConsumer, times(1)).resume(
+			any(io.vertx.kafka.client.common.TopicPartition.class), any());
+		verify(kafkaConsumer, times(1)).commit(anyMap(), any());
+		verify(kafkaProducer, times(0)).write(any(), any());
+	}
+
+	private KafkaConsumerRecords<String, String> prepareKafkaConsumerRecords(int recordCount) {
+		List<ConsumerRecord<String,String>> consumerRecordList = new ArrayList<>();
+		for(int i=0; i<recordCount; i++)
+			consumerRecordList.add(
+				new ConsumerRecord<String,String>(
+					MessageBusAddress.PACKET_VALIDATOR_BUS_IN.getAddress(), 0, i, "1000"+i,
+					"{\"rid\":\"1000"+i+"\", \"reg_type\": \"NEW\" }"));
+		Map<TopicPartition, List<ConsumerRecord<String,String>>> topicPartitionConsumerRecordListMap = 
+			new HashMap<TopicPartition, List<ConsumerRecord<String,String>>>();
+		topicPartitionConsumerRecordListMap.put(
+			new TopicPartition(MessageBusAddress.PACKET_VALIDATOR_BUS_IN.getAddress(), 0), 
+				consumerRecordList);
+		ConsumerRecords<String,String> consumerRecords = 
+			new ConsumerRecords<String,String>(topicPartitionConsumerRecordListMap);
+		KafkaConsumerRecords<String, String> kafkaConsumerRecords = 
+			new KafkaConsumerRecordsImpl<String, String>(consumerRecords);
+		return kafkaConsumerRecords;
+	}
+
+	@Test
+	public void testConsumerHealthCheck(TestContext testContext) {
+		kafkaMosipEventBus = new KafkaMosipEventBus(vertx, "localhost:9091", "group_1", "batch", "100", 60000,
+				eventTracingHandler);
+		final Async async = testContext.async();
+		Handler<HealthCheckDTO> eventHandler = Mockito.mock(Handler.class);
+		AsyncResult<Map<String, List<PartitionInfo>>> asyncResult = Mockito.mock(AsyncResult.class);
+		Mockito.when(asyncResult.succeeded()).thenReturn(true);
+		doAnswer((Answer<AsyncResult<Map<String, List<PartitionInfo>>>>) arguments -> {
+			((Handler<AsyncResult<Map<String, List<PartitionInfo>>>>) arguments.getArgument(0))
+					.handle(asyncResult);
+			if (!async.isCompleted())
+				async.complete();
+			return null;
+		}).when(kafkaConsumer).listTopics(any());
+		kafkaMosipEventBus.consumerHealthCheck(eventHandler, MessageBusAddress.PACKET_VALIDATOR_BUS_IN.toString());
+		async.await();
+		verify(kafkaConsumer, times(1)).listTopics(any());

```

Comment:
We should also verify if the eventHander is called once and if the health check dto is giving event bus connected as true.

With the same you can also create a 2 negative test case, where this is false 
1. if the listTopics is throwning a exception
2. if list topics is timing out (our original scenario)

Repeat the same for sender health check.

Reply:
Fixed